pkg/edit/config_api.go

```diff
@@ -60,6 +61,20 @@ func initAfterReadline(appSpec *cli.AppSpec, ev *eval.Evaler, nb eval.NsBuilder)
 	})
 }
 
+//elvdoc:var after-command
+//
+// A list of functions to call after each interactive command completes. Each function is
+// called with a single string argument containing the code that has been read.
+
+func initAfterCommand(appSpec *cli.AppSpec, ev *eval.Evaler, nb eval.NsBuilder) {
+	hook := newListVar(vals.EmptyList)
+	nb["after-command"] = hook
+	appSpec.AfterCommand = append(appSpec.AfterCommand,
+		func(line string, duration float64, err error) {
+		callHooks(ev, "$<edit>:after-command", hook.Get().(vals.List), line, duration, err)
```

Comment:
Instead of calling the after-command hooks with 3 arguments, better to call it with one map argument [&executed-code=... &duration=... &error=...]

Reply:
Good idea since it future proofs the API against additional attributes. Will change. However, the cost of this mechanism is another reason to retain a `$edit:command-duration` variable as a close to zero cost, both edit and run time, for users that simply want to display the duration in their prompt.